### PR TITLE
video: add logo recognition samples

### PR DESCRIPTION
**Pull request by sirtorry**

### CODE REVIEW

File: videointelligence/annotate/logo_detection.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+func logoDetection(w io.Writer, filename string) error {
+	// filename := "resources/googlework_tiny.mp4"
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := video.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("video.NewClient: %v", err)
+	}
```
Comment (shollyman) [inline videointelligence/annotate/logo_detection.go]: defer client.Close() after the err check to clean up the instantiated client, here and in the other sample.

File: videointelligence/annotate/logo_detection.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		return fmt.Errorf("AnnotateVideo: %v", err)
+	}
+
+	resp, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("Wait: %v", err)
+	}
+
+	// Only one video was processed, so get the first result.
+	result := resp.GetAnnotationResults()[0]
```
Comment (shollyman) [inline videointelligence/annotate/logo_detection.go]: Will you ever get no results in the success case?

File: videointelligence/annotate/logo_detection.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package annotate contains speech examples.
+package annotate
+
+// [START video_detect_logo]
+
```
Comment (tbpg) [inline videointelligence/annotate/logo_detection.go]: Delete this line.

File: videointelligence/annotate/logo_detection.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	// Only one video was processed, so get the first result.
+	result := resp.GetAnnotationResults()[0]
+
+	// Annotations for list of logos detected, tracked and recognized in video.
+	for _, annotation := range result.LogoRecognitionAnnotations {
+		fmt.Fprintf(w, "Description: %q\n", annotation.Entity.GetDescription())
+		// Opaque entity ID. Some IDs may be available in [Google Knowledge
+		// Graph Search API](https://developers.google.com/knowledge-graph/).
```
Comment (tbpg) [inline videointelligence/annotate/logo_detection.go]: Markdown links unfortunately don't work.
```suggestion
		// Opaque entity ID. Some IDs may be available in Google Knowledge
		// Graph Search API (https://developers.google.com/knowledge-graph/).
```

File: videointelligence/annotate/logo_detection_test.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		var buf bytes.Buffer
+		if err := logoDetection(&buf, file); err != nil {
+			r.Errorf("logoDetection: %v", err)
+			return
+		}
+
+		wants := [3]string{"Description", "Confidence", "Segment"}
+		for _, want := range wants {
+			if got := buf.String(); !strings.Contains(got, want) {
+				r.Errorf(`logoDetection(%q): got %q, want %q`, file, got, want)
```
Comment (tbpg) [inline videointelligence/annotate/logo_detection_test.go]:
```suggestion
				r.Errorf("logoDetection(%q): got %q, want %q", file, got, want)
```

File: videointelligence/annotate/logo_detection_gcs_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		var buf bytes.Buffer
+		if err := logoDetectionGCS(&buf, gcsURI); err != nil {
+			r.Errorf("logoDetectionGCS: %v", err)
+			return
+		}
+
+		wants := [3]string{"Description", "Confidence", "Segment"}
+		for _, want := range wants {
+			if got := buf.String(); !strings.Contains(got, want) {
+				r.Errorf(`logoDetection(%q): got %q, want %q`, gcsURI, got, want)
```
Comment (tbpg) [inline videointelligence/annotate/logo_detection_gcs_test.go]:
```suggestion
				r.Errorf("logoDetection(%q): got %q, want %q", gcsURI, got, want)
```

File: videointelligence/annotate/logo_detection.go (hunk 2 above)
Comment (sirtorry) [reply videointelligence/annotate/logo_detection.go]: don't think so

**Pull request merged**